### PR TITLE
Env in transform

### LINKED ISSUE #668: And an Env section to Transform
For defining environment variables in case that's not obvious from the title.

### PR DESCRIPTION
**Pull request by jdoliner**
Fixes #668.

### CODE REVIEW

**Comment by jdoliner** [comment]
@sjezewski whenever you have a chance.

File: src/server/pachyderm_test.go [hunk 1]
```diff
@@ -2189,6 +2189,44 @@ func TestPipelineInfoDestroyedIfRepoCreationFails(t *testing.T) {
...
+	// Do first commit to repo
+	commit, err := c.StartCommit(dataRepo, "", "")
+	require.NoError(t, err)
+	_, err = c.PutFile(dataRepo, commit.ID, "file", strings.NewReader("foo\n"))
+	require.NoError(t, err)
+	require.NoError(t, c.FinishCommit(dataRepo, commit.ID))
+	commitInfos, err := c.FlushCommit([]*pfsclient.Commit{commit}, nil)
+	require.Equal(t, 1, len(commitInfos))
+	var buffer bytes.Buffer
+	require.NoError(t, c.GetFile(pipelineName, commitInfos[0].Commit.ID, "file", 0, 0, "", nil, &buffer))
```
Comment (sambooo) [inline src/server/pachyderm_test.go]: GetFile is failing here because of the changes in #663 I think

**Comment by sjezewski** [comment]
Is there a way we can test the 'Secrets' feature as well?

Also - now that we have a place for docs, we need to make sure to document any new features (or they basically don't exist for users). This would probably go [here](https://github.com/pachyderm/pachyderm/blob/master/doc/pipeline_spec.md) I think

**Comment by jdoliner** [comment]
Alright this now has tests for secrets as well as an implementation for them.
Ready for another look I think.

File: src/server/pachyderm_test.go [hunk 2]
```diff
@@ -2189,6 +2189,71 @@ func TestPipelineInfoDestroyedIfRepoCreationFails(t *testing.T) {
...
+	// create pipeline
+	pipelineName := uniqueString("pipeline")
+	_, err = c.PpsAPIClient.CreatePipeline(
+		context.Background(),
+		&ppsclient.CreatePipelineRequest{
+			Pipeline: client.NewPipeline(pipelineName),
+			Transform: &ppsclient.Transform{
+				Cmd: []string{"sh"},
+				Stdin: []string{
+					"ls /var/secret",
```
Comment (sjezewski) [inline src/server/pachyderm_test.go]: I think this was a line for debugging the test. I don't think we need this anymore for the purposes of what we're testing.

**Comment by sjezewski** [comment]
Minor comment in the test, but the coverage looks good.

We still need to add these features to the pipeline spec documentation though.

**Comment by jdoliner** [comment]
@sjezewski another look? I think this is ready to go.

**Comment by jdoliner** [comment]
Really just a meme I guess, all I did was add it to docs.

**Comment by sjezewski** [comment]
image

**Comment by jdoliner** [comment]
Meme lgtm.

**Pull request merged**